test/test_query.cpp

```diff
@@ -3402,64 +3400,64 @@ TEST(Query_Float)
     // ------ Test sum()
     // ... NO conditions
     double sum1_d = 2.20 + 2.21 + 2.22 + 2.20 + 3.20;
-    CHECK_APPROXIMATELY_EQUAL(sum1_d, t.where().sum_double(col_double), 10 * epsilon);
+    CHECK_APPROXIMATELY_EQUAL(sum1_d, t.where().sum(col_double)->get_double(), 10 * epsilon);
 
     // Note: sum of float is calculated by having a double aggregate to where each float is added
     // (thereby getting casted to double).
     double sum1_f = double(1.10f) + double(1.13f) + double(1.13f) + double(1.10f) + double(1.20f);
-    double res = t.where().sum_float(col_float);
+    double res = t.where().sum(col_float)->get_double();
     CHECK_APPROXIMATELY_EQUAL(sum1_f, res, 10 * epsilon);
 
     // ... with conditions
     double sum2_f = double(1.13f) + double(1.20f);
     double sum2_d = 2.21 + 3.20;
     Query q2 = t.where().between(col_float, 1.13f, 1.20f).not_equal(col_double, 2.22);
-    CHECK_APPROXIMATELY_EQUAL(sum2_f, q2.sum_float(col_float), 10 * epsilon);
-    CHECK_APPROXIMATELY_EQUAL(sum2_d, q2.sum_double(col_double), 10 * epsilon);
+    CHECK_APPROXIMATELY_EQUAL(sum2_f, q2.sum(col_float)->get_double(), 10 * epsilon);
```

Comment:
It may be unintuitive for users of this API to know that the sum of a float column is a double. This is something that was baked into the return type previously, but now has to be known. I think the tradeoffs are still heavily in favour of this change overall, but this is the cost of using a Mixed type for everything.

Reply:
I don't like the sum of floats being a double for a few reasons, but I think in practice SDKs just switch on the Mixed's type anyway so returning any type in the Mixed probably just works out.

Reply:
If SDKs are dynamically switching on the Mixed's type that would be ideal actually since it leaves room for us to change the underlying type in the future in a less breaking way. I vaguely recall some desire to use Decimal128 for average/sum of float/double, but I think that was stalled because of some non-trivial tradeoffs.